### DISCUSSION TITLE
Mesh setting up message at the beginning of output

### DISCUSSION THREAD

**Discussion by xueyang94** [idaholab/moose#20940] [Q&A Meshing] [answered]
I sometimes randomly (but rarely) receive this message at the beginning of the output when running a moose input file. With this message, the simulation takes a long time to run. But when I ran the same input file for a second time, I don't receive the mesh setting message and everything is normal. Where does this message comes from and how to make sure the future run does not have this message? Thanks. -Xueyang
![image](https://user-images.githubusercontent.com/31705995/166482428-156f56f1-b306-4b2e-b702-253a61f18835.png)

**Comment by GiudGiud** [comment]
Hello

These are messages from the live perf graph. It's outputting to let you know it's not stuck. 
https://mooseframework.inl.gov/source/utils/PerfGraph.html
I imagine the Reading mesh ... is always there and the Updating mesh ... message is only sometimes there?
Are you running with a distributed mesh?

Guillaume

**Reply by xueyang94** [reply]
The Reading mesh and Updating mesh either appear together or don't appear at all. In the latter case the output just starts with Framework Information. I am not using distributed mesh. The mesh is read from an exodus file and the variables are imported. This is strange because I am running a bunch of input files with everything else exactly the same except for one diffusion material property parameter. Within those input files, only one gave me the Setting Mesh info I showed above, and that one took much longer wall time. But I ran that one input again, it didn't show the mesh setting message and finished normally just like other files.

**Reply by GiudGiud** [reply]
Is this on an HPC platform? Or a work computer?
These logs appearing/disappearing is tied to how fast you are loading the mesh basically. So if something is slowing down file IO outside of MOOSE that can trigger that.
Reading mesh in MOOSE is deterministic otherwise.

**Reply by xueyang94** [reply]
This particular batch of files was on HPC, but I have seen the same thing happened before on local machine. I don't think these logs appearing is an issue, the issue is that in all of the following iterations, each step takes far more wall time somehow. If something was slowing down the IO when I started the simulation, I would expect the simulation speed to catch up after a while. By ending the simulation and restarting immediately, the normal speed was recovered.

**Reply by GiudGiud** [reply]
so the whole simulation is slower when they appear?

**Reply by xueyang94** [reply]
Correct.

**Reply by GiudGiud** [reply]
if you can share your inputs I can see if I can reproduce this. But the fact that this affects the whole simulation and not parts of it points to an issue with either your machine or how moose is being ran rather than a buggy feature which would only be present in parts of the simulation.
Are you running threads or MPI?

**Reply by xueyang94** [reply]
I'm running MPI. On both cluster and my local machine, with that message, the simulation is slowed down. My input file imports mesh from an exodus file which is 25GB. How would you prefer me to share both the exodus and input file? Thanks for your help.

**Reply by GiudGiud** [reply]
That message appears because the simulation is slow I think, not the other way around.
With MPI a classic 2x slowdown is if two processes get bound to the same core. Or if you are using all the cores but one of the cores is busy doing something else for the system (often an issue on HPC).

How many processes are you using and how many cores are there on the machines?
If you can share those I can see if it's reproducible on INL HPC.

**Reply by xueyang94** [reply]
On my university HPC, for one simulation I request 8 nodes and 32 MPI processes. Each node has 2 CPU and 32 cores. So I use 16 CPU and 256 cores in total.

My local machine has 1 CPU and 8 cores in total. I use all cores to execute a simulation: `mpirun -np 8 -opt -i`.

**Reply by GiudGiud** [reply]
So you could send me the inputs by email. My email is on my github profile. 
What application do I need to run it? just moose?
I ll see if I can reproduce the slowdown.

**Comment by GiudGiud** [comment, marked as answer]
@xueyang94 any luck finding the culprit?

**Reply by xueyang94** [reply]
@GiudGiud I have emailed you back days ago. The printing of mesh setting messages and the subsequent slow down is most likely due to the machine being busy. Thanks for the help.

**Reply by GiudGiud** [reply]
oh sorry. Glad this is settled and thanks for letting me know
